aspnetcore/fundamentals/servers/kestrel/http3.md

```diff
@@ -12,34 +12,49 @@ uid: fundamentals/servers/kestrel/http3
+[HTTP/3](https://quicwg.org/base-drafts/draft-ietf-quic-http.html) is the third and upcoming major version of HTTP. HTTP/3 uses the same semantics as HTTP/1.1 and HTTP/2: the same request methods, status codes, and message fields apply to all versions. The differences are in the underlying transport. Both HTTP/1.1 and HTTP/2 use TCP as their transport. HTTP/3 uses a new transport technology developed alongside HTTP/3 called [QUIC](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-34).
 
-## Requirements
+HTTP/3 and QUIC have a number of benefits compared to HTTP/1.1 and HTTP/2:
 
-HTTP/3 is not supported everywhere. The requirements are different depending on the operating system that Kestrel is running on.
+* Faster response time of the first request. QUIC and HTTP/3 negotiates the connection in fewer round-trips between the client and the server. The first request reachs the server faster.
+* Improved experience when there is connection packet loss. HTTP/2 multiplexes multiple requests via one TCP connection. Packet loss on the connection affects all requests. This problem is called "head-of-line blocking". Because QUIC provides native multiplexing, lost packets only impact the requests where data has been lost.
+* Supports transitioning between networks. This feature is useful for mobile devices where it is common to switch between WIFI and cellular networks as a mobile device changes location. Currently HTTP/1.1 and HTTP/2 connections fail with an error when switching networks. An app or web browsers must retry any failed HTTP requests. HTTP/3 allows the app or web browser to seamlessly continue when a network changes. Kestrel doesn't support network transitions in .NET 6. It may be available in a future release.
 
-Kestrel enables HTTP/3 only on environments that support it. That means it's possible to configure a port to support all HTTP protocols. For example, `HttpProtocols.Http1AndHttp2AndHttp3`, and Kestrel's HTTP/3 is available on environments where it's supported.
+> [!IMPORTANT]
+> HTTP/3 is available in .NET 6 as a preview feature. The HTTP/3 specification hasn't been finalized. Additionally, there may be behavioral or performance issues in HTTP/3 with .NET 6.
+>
+> Apps that want take advantage of HTTP/3 should be designed to support all 3 versions versions. If issues are identified in HTTP/3 then it is recommended that apps temporarily disable it until issues are resolved and the app can be updated.
+
+## HTTP/3 requirements
+
+HTTP/3 isn't supported everywhere. The requirements are different depending on the operating system hosting Kestrel.
+
+When using HTTP/3, we recommend enabling it and the other HTTP versions. Because Kestrel only enables HTTP/3 if an environment supports it, a port configured to serve all HTTP protocols automatically enable HTTP/3 if it is available. If HTTP/3 isn't available then the port continues to serve HTTP/1.1 and HTTP/2 requests.
```

Comment:
This parts a little confusing. Kestrel doesn't automatically enable HTTP/3, you have to turn it on. It will no-op on platforms where it's not supported if a fallback is available.

Reply:
It's also important to have a fallback protocol because an application is likely to experience connectivity issues for QUIC / HTTP/3 until all routers, firewalls, and proxies are updated to support it.